tss463_van.cpp

```diff
@@ -268,25 +281,29 @@ uint8_t TSS463_VAN::registers_get(uint8_t address, volatile uint8_t values[], ui
 
     TSS463_SELECT();
 
-    _delay_ms(4);
+    delayMicroseconds(1);//at 8MHZ max speed (4 clocks XTAL)
+    //_delay_ms(4);
     //At the beginning of a transmission over the serial interface, the first byte is the address of the TSS463C register to be accessed
     value = spi_transfer(address);
     if (value != ADDR_ANSW)
         error++;
-    _delay_ms(8);
+    delayMicroseconds(2);//at 8MHZ max speed (8 clocks XTAL)
+    //_delay_ms(8);
     //The next byte transmitted is the control byte that determines the direction of the communication
     value = spi_transfer(READ);
     if (value != CMD_ANSW)
         error++;
-    _delay_ms(15);
+    delayMicroseconds(4);//at 8MHZ max speed (15 clocks XTAL)
+    //_delay_ms(15);
     //When the master (CPU) conducts a read, it sends an address byte, a control byte and dummy characters (�0xFF�for instance) on its MOSI line
 
     // TSS463 has auto-increment of address-pointer
     for (uint8_t i = 0; i < count; i++)
     {
         values[i] = spi_transfer(0xff);
+		delayMicroseconds(3);//at 8MHZ max speed (12 clocks XTAL)
```

Comment:
Based on the datasheet it isn't needed to wait after every byte in the for loop. It would be enough to wait after the transmit finished (outside the loop).

Reply:
This part of the code was because of this sentence "the minimum inter-character delay is 12 µs (12 oscillator periods)" in the topic "Clocks and Speed ​​Considerations" on the datasheet, but as English is not my primary language,  maybe I got it wrong.
I noticed it does work with the delay outside the loop, but I´m not sure if that´s correct.

Reply:
Well I am not sure about that either. I'm not a native English speaker, it was just an assumption from my side. The diagram at Figure 7 shows the Data part as one unit and not a series of bytes, with delays between them. I think this is kind of misleading and maybe you are right.